compiler-core/src/build/project_compiler.rs

```diff
@@ -112,6 +119,11 @@ where
 
     /// Returns the compiled information from the root package
     pub fn compile(&mut self) -> Result<Package> {
+        let lock_guard = self.lock.try_lock();
+        if(lock_guard.is_err()) {
+            return lock_guard // TODO fix return type
```

Comment:
Is there a better way to return this? Or am I doing something wrong?

Reply:
Use `lock` rather than `try_lock` :)